Migrate airflow-great-expectations guide (#1076)
astronomer/docs: guides/airflow-great-expectations.md

---
title: "Integrating Airflow and Great Expectations"

> sidebar_label: "Integrating Airflow and Great Expectations"

Reviewer: ```suggestion
sidebar_label: "Great Expectations"
```

> description: "Using the Great Expectations provider natively in your Airflow DAGs."

Reviewer: ```suggestion
description: "Use the Great Expectations provider in your Airflow DAGs."
```

id: airflow-great-expectations

> tags: ["DAGs", "Integrations"]

Reviewer: ```suggestion
```

---

> > You can now find the [Great Expectations Provider](https://registry.astronomer.io/providers/great-expectations) on the [Astronomer Registry](https://registry.astronomer.io), the discovery and distribution hub for Apache Airflow integrations created to aggregate and curate the best bits of the ecosystem.

Reviewer: ```suggestion
:::info

You can now find the [Great Expectations Provider](https://registry.astronomer.io/providers/great-expectations) on the [Astronomer Registry](https://registry.astronomer.io), the discovery and distribution hub for Apache Airflow integrations created to aggregate and curate the best bits of the ecosystem.

:::
```

## Setup

This guide assumes that you have downloaded the code from the [demo repository](https://github.com/astronomer/airflow-data-quality-demo/), which contains a sample Great Expectations project.

If you want to use your own Great Expectations project along with this guide, ensure you have completed the following steps:

- Initialized a Great Expectations project
- Configured at least one Datasource `my_datasource`
- Created at least one Expectation Suite `my_suite`
- Created a Checkpoint `my_checkpoint`

If you set up a project manually, you will see a `great_expectations` directory which contains several sub-directories, as well as the `great_expectations.yml` configuration file. If you cloned the demo repository, the `great_expectations` directory can be found under `include/`.

> Note: The `GreatExpectationsOperator` requires Airflow 2.1 or later, and you will need to change the value of `enable_xcom_pickling` to `true` in your airflow.cfg. If you are using an Astronomer project structure, add `ENV AIRFLOW__CORE__ENABLE_XCOM_PICKLING=True` to your Dockerfile. If you are working from the demo repository, this step has already been completed for you.

> ## Use Case: Great Expectations Operator

Reviewer: ```suggestion
## Use Case: Great Expectations operator
```

[Checkpoints](https://docs.greatexpectations.io/docs/reference/checkpoints_and_actions) provide a convenient abstraction for bundling the validation of a Batch (or Batches) of data against an Expectation Suite (or several), as well as the actions that should be taken after the validation.

Our [demo repository](https://github.com/astronomer/airflow-data-quality-demo/) uses the following configuration:

- The `great_expectations` directory is accessible by your DAG, as it is loaded into Docker as part of the `include` directory.
- The Great Expectations provider is installed when you run `astro dev start`, as it is part of `requirements.txt`. For Astronomer projects, a provider version >=`0.1.1` is required. If you are not using an Astronomer environment, install the Great Expectations provider in your environment manually:

    ```bash
    pip install great_expectations airflow-provider-great-expectations>=0.1.0
    ```

    It’s recommended that you specify a version when installing the package. To make use of the latest Great Expectations V3 API, you need to specify a version >= `0.1.0`.

> ### Using the Great Expectations Operator

Reviewer: ```suggestion
### Using the Great Expectations operator
```

For more information about possible parameters and examples, see the [README in the provider repository](https://github.com/great-expectations/airflow-provider-great-expectations) and the [example DAG in the provider package](https://registry.astronomer.io/dags/example-great-expectations-dag).

### Connections and backends

The `GreatExpectationsOperator` can run a checkpoint on a dataset stored in any backend compatible with Great Expectations. For example, BigQuery, MSSQL, MySQL, PostgreSQL, Redshift, Snowflake, SQLite, Athena, and so on. All that’s needed to get the Operator to point at an external dataset is to set up an [Airflow Connection](https://www.astronomer.io/guides/connections) to the `datasource`, and add the connection to your Great Expectations project [using the CLI to add a Postgres backend](https://docs.greatexpectations.io/docs/guides/connecting_to_your_data/database/postgres). Then, if using a `DataContextConfig` or `CheckpointConfig`, ensure that the `"datasources"` field refers to your backend connection name.

> ## Conclusion

Reviewer: ```suggestion
## Next steps
```
